tests/select_for_update/tests.py

```diff
@@ -322,3 +395,99 @@ def test_ordered_select_for_update(self):
         with transaction.atomic():
             qs = Person.objects.filter(id__in=Person.objects.order_by('-id').select_for_update())
             self.assertIn('ORDER BY', str(qs.query))
+
+    def run_select_for_update_with_related_objects_scenario(self, of):
```

Comment:
These tests are a bit complicated -- do you think they offer substantial value? I didn't check the origin of the existing threading tests but I feel like it's testing the database more than Django. If Django generates the correct query, is there something else that could go wrong?